### DISCUSSION TITLE
Canceling visit event

### DISCUSSION THREAD

**Discussion by oidre** [inertiajs/inertia#500] [Help] [answered]
I have tested using 2 routes, the first route processing time almost instantly and the second one using sleep function for simulating delay request, it takes 5 seconds.
I try clicking the second route twice then clicking the first route, it takes 10 seconds before reaching  the first route. How to cancel the request made from clicking twice the second route? So the processing time should be instantly because only visit the first route.

**Comment by ajnsn** [comment, marked as answer]
@oidre Please take a look at the documentation. Maybe that already solves your problem:

[Visit cancellation](
https://inertiajs.com/manual-visits#visit-cancellation)

**Reply by oidre** [reply]
I think that's not possible because the documentation said cancelling event only work on some event like **before**, **invalid**, and **error**.
The problem is visit successfull but delay because network issue, so it's not cancellable and still stacking and the latest visit must wait the previous visit till finish :/

**Reply by ajnsn** [reply]
@oidre There's also a part about visit cancellation. I am curious if that does work?

```js
Inertia.post('/users', data, {
  onCancelToken: (cancelToken) => (this.cancelToken = cancelToken),
})

// Cancel the visit
this.cancelToken.cancel()
```

**Reply by oidre** [reply]
@ajnsn That's just cancelling the visit in the UI.
I have been tried clicking the same route twice which takes 5 seconds each, so it takes 10 seconds to complete, then cancelling the visit using cancel token.
The progress bar stopped and not redirected to that route, but when i try another route that takes time almost instantly, it should be instant but the result still waiting previous visit to be done (10 seconds)

**Reply by ajnsn** [reply]
@oidre Inertia can only cancel visits (requests) within your frontend. There's no "wire" between the frontend and backend which tells the backend to cancel an action. 

Afaik Inertia visits are synchronous, like a page load. If your action takes so long, you may want to use an asynchronous xhr/fetch request instead.

**Reply by oidre** [reply]
@ajnsn Ahh I see, thank you for explaining.
